pandas/io/common.py

```diff
@@ -805,6 +805,19 @@ def __init__(
         # _PathLike[str]], IO[bytes]]"
         super().__init__(file, mode, **kwargs_zip)  # type: ignore[arg-type]
 
+    # GH39465
+    # If an explicit archive_name is not given, we still want the file _inside_ the zip
+    # file _not_ to be named something.zip, because that causes massive confusion.
+    def infer_filename(self):
+        if isinstance(self.filename, (os.PathLike, str)):
+            filename = Path(self.filename)
+            if filename.suffix == ".zip":
```

Comment:
are all of these cases hit in tests?

Reply:
(https://github.com/pandas-dev/pandas/pull/44445/commits/c1c89719d1097ae3603f6b969fd0da483bfa8dc0) covers a few more scenarios.

The only case that is not covered at the moment is that where `self.filename` is neither `os.PathLike` nor `str`.
Looking at the super class, I don't see how this could ever happen. But I believe it's still good to keep the check in on our side.